src/pages/index.js

```diff
@@ -29,50 +29,165 @@ import {
+}
+
+const deleteCardPopup = new PopupWithConfirmation("#delete-card-modal");
+deleteCardPopup.setEventListeners();
+
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.setLoading(true);
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.remove();
+        deleteCardPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        deleteCardPopup.setLoading(false, "Yes");
+      });
+  });
+}
+
+function createCard({ name, link, isLiked, _id, userId, ownerId }) {
   const cardElement = new Card(
-    { name, link },
+    { name, link, isLiked, _id, userId, ownerId },
+    userID,
     "#card-template",
-    ({ name, link }) => {
-      previewImagePopup.open({ name, link });
-    }
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick
   );
   return cardElement.getView();
 }
-
+// =============================================================================
 // Preview Popup
+// =============================================================================
 
 const previewImagePopup = new PopupWithImage({
   popupSelector: "#preview-modal",
+  handleFormSubmit: handleProfileEditSubmit,
 });
 previewImagePopup.setEventListeners();
 
-//Profile Popup
-
-const newUser = new UserInfo(".profile__title", ".profile__description");
+// =============================================================================
+// Profile Popup
+// =============================================================================
 
-function handleProfileEditSubmit({ title, description }) {
-  newUser.setUserInfo(title, description);
+function handleProfileEditSubmit({ name, about }) {
+  newUser.setUserInfo(name, about);
   profilePopup.close();
 }
 
-const profilePopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
+const profilePopup = new PopupWithForm({
+  popupSelector: "#profile-edit-modal",
+  handleFormSubmit: (inputValues) => {
+    profilePopup.showLoading();
+    api
+      .editProfile(inputValues)
+      .then(() => {
+        userInfo.setUserInfo(inputValues);
+        profilePopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        profilePopup.hideLoading();
+      });
+  },
+  loadingButtonText: "Saving...",
+});
+
 profilePopup.setEventListener();
 
 profileEditButton.addEventListener("click", () => {
-  const userData = newUser.getUserInfo();
+  const userData = userInfo.getUserInfo();
   modalTitleInput.value = userData.name;
-  modalDescriptionInput.value = userData.profession;
+  modalDescriptionInput.value = userData.about;
   profilePopup.open();
 });
 
+// =============================================================================
 // Validation
+// =============================================================================
 
 const addFormValidator = new FormValidator(formValidationConfig, addCardForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```